examples/cpp/helloworld/greeter_server_ssl.cc

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright 2022 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+#include <memory>
+#include <string>
+#include <sstream>
+#include <fstream>
+#include <iostream>
+
+#include <grpcpp/ext/proto_server_reflection_plugin.h>
+#include <grpcpp/grpcpp.h>
+#include <grpcpp/health_check_service_interface.h>
+
+#ifdef BAZEL_BUILD
+#include "examples/protos/helloworld.grpc.pb.h"
+#else
+#include "helloworld.grpc.pb.h"
+#endif
+
+using grpc::Server;
+using grpc::ServerBuilder;
+using grpc::ServerContext;
+using grpc::Status;
+
+using helloworld::HelloRequest;
+using helloworld::HelloReply;
+using helloworld::Greeter;
+
+// Logic and data behind the server's behavior.
+class GreeterServiceImpl final : public Greeter::Service
+{
+  Status SayHello ( ServerContext* context,
+                    const HelloRequest* request,
+                    HelloReply* reply ) override
+  {
+    std::string prefix { "Hello " };
+
+    reply->set_message ( prefix + request->name () );
+
+    return Status::OK;
+  }
+};
+
+void read ( const std::string& filename, std::string& data )
```

Comment:
consider sharing this as a util target rather than duplicating the code.

Reply:
Done